libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -239,6 +239,11 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
                     // In the case of Invoke, just create a fake one. Match the incoming activityId if it's there. 
                     response = new ResourceResponse(activity.Id ?? string.Empty);
                 }
+                else if (activity.Type == ActivityTypes.Trace && activity.ChannelId != "emulator")
```

Comment:
Do you think limiting to *ONLY* the emulator is correct? That seems like a great default, but I can envision scenarios where I would want to the trace send to other channels as well.